analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -781,6 +781,47 @@ public void PropertyReference_Write_SetsNotNull()
             validator.ValidateTag("AfterIndexer", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
 
+        [TestMethod]
+        public void PropertyReference_Write_SetsNull_AndRead_SetsNotNull_ForAutoProperties()
+        {
+            const string code = """
+                public object AutoProperty { get; set; }
+
+                public void M()
+                {
+                    AutoProperty = null;
+                    Tag("AfterSetNull", AutoProperty);
+                    AutoProperty.ToString();
+                    Tag("AfterReadReference", AutoProperty);
+                }
+                """;
+            var validator = SETestContext.CreateCSMethod(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.PropertyReference);
+            validator.ValidateTag("AfterSetNull", x => x.Should().HaveNoConstraints()); // FN Auto-properties should behave the same as fields. Expected: HaveOnlyConstraint(ObjectConstraint.Null)
+            validator.ValidateTag("AfterReadReference", x => x.Should().HaveNoConstraints()); // FN Expected: HaveOnlyConstraint(ObjectConstraint.NotNull)
+        }
+
+        [TestMethod]
+        public void PropertyReference_NotTrackedFor_FullProperties()
```

Comment:
To explain the context and then say the expectations
```suggestion
        public void PropertyReference_FullProperties_NotTracked()
```